py/desispec/desi_proc_dashboard.py

```diff
@@ -0,0 +1,383 @@
+                hs.write(strFailed)
+                hs.close()
+            strTable=strTable+str_row
+        strTable=strTable+"</table>"
+        return strTable
+
+    def _add_js_script1(self):
+        s="""<script>
+            var coll = document.getElementsByClassName('collapsible');
+            var i;
+            for (i = 0; i < coll.length; i++) {
+                coll[i].nextElementSibling.style.maxHeight='0px';
+                coll[i].addEventListener('click', function() {
+                    this.classList.toggle('active');
+                    var content = this.nextElementSibling;
+                    if (content.style.maxHeight){
+                       content.style.maxHeight = null;
+                    } else {
+                      content.style.maxHeight = '0px';
+                            } 
+                    });
+             };
+             var b1 = document.getElementById('b1');
+             b1.addEventListener('click',function() {
+                 for (i = 0; i < coll.length; i++) {
+                     coll[i].nextElementSibling.style.maxHeight=null;
+                                                   }});
+             var b2 = document.getElementById('b2');
+             b2.addEventListener('click',function() {
+                 for (i = 0; i < coll.length; i++) {
+                     coll[i].nextElementSibling.style.maxHeight='0px'
+                             }});
+            </script>"""
+        return s
+
+    def _add_js_script2(self,n_modal):
+        s="""<script>"""
+        for i in range(n_modal):
+            s=s+"""
+                var modal"""+str(i)+""" = document.getElementById('modal"""+str(i)+"""');
+                var l"""+str(i)+""" = document.getElementById('Btn"""+str(i)+"""');
+
+                l"""+str(i)+""".addEventListener('click',function() {
+                  modal"""+str(i)+""".style.display = "block";
+                })
+
+                span"""+str(i)+""".addEventListener('click',function() {
+                  modal"""+str(i)+""".style.display = "none";
+                })"""
+
+        s=s+"""</script>"""
+        return s
+
+
+
+    def what_night_is_it(self):
+        d = datetime.datetime.utcnow() - datetime.timedelta(7/24+0.5)
+        tonight = int(d.strftime('%Y%m%d'))
+        return tonight
+
+    def find_newexp(self,night, fileglob, known_exposures):
+        datafiles = sorted(glob.glob(fileglob))
+        newexp = list()
+        for filepath in datafiles:
+            expid = int(os.path.basename(os.path.dirname(filepath)))
+            if (night, expid) not in known_exposures:
+                newexp.append( (night, expid) )
+        return set(newexp)
+    def check_running(self):
+        a=psutil.process_iter()
+        running='No'
+        for p in a:
+            if 'desi_dailyproc' in ' '.join(p.cmdline()):
+                running='Yes'
+        return running
+
+        
+if __name__=="__main__":
+    process=DESI_PROC_DASHBOARD()
```

Comment:
Please add a wrapper script in `desispec/bin/desi_proc_dashboard` (no .py on the end) that just loads and calls this code, so that when desispec is installed that script will be in the path.